HACKING.md

```diff
@@ -0,0 +1,95 @@
+# Hacking
+
+## Quickstart
+
+Use [ghcup][ghcup] to install the `cabal` cli tool and the ghc version we're using:
+
+```sh
+$ ghcup install ghc 8.10
+$ ghcup set ghc 8.10
+$ ghcup install cabal 
+$ ghcup set cabal
+$ cabal update
+$ cabal build
+```
+
+### Building
+
+In the base directory, run `cabal build`
+
+### Running tests
+
+In the base directory, run `cabal test`
+
+## Tools
+
+| name | description |
+| ---- | ----------- |
+| [ghcup][ghcup] | Used to manage installed versions of ghc and cabal-install |
+| ghc | The haskell compiler (installed via ghcup) |
+| cabal-install | The package manager we use (installed via ghcup) |
+| [haskell-language-server][hls] | LSP server for haskell projects |
+| [hlint][hlint] | A linting + hints tool for haskell code. It provides really useful suggestions |
+| [ormolu][ormolu] | A haskell source code formatter |
+| [fourmolu][fourmolu] | A forked version of ormolu that we are evaluating |
+
+### Installing haskell-language-server
+
+In VSCode: Install the "Haskell Language Server" (`haskell.haskell`) plugin in VSCode. In the
+settings there's a `Language Server Haskell: Hie Variant` option, which you can
+set to `haskell-language-server`
+
+## Linting
+
+Install [hlint][hlint].  Run `hlint /path/to/file-or-directory`.  You can also configure the haskell language server to run hlint.
```

Comment:
We should exclusively recommend that folks use hlint via haskell-language-server -- because it's built-in and enabled by default

Additionally, hlint chokes on some source files if it's not configured correctly (which hls does for us)